wolfcrypt/src/port/Renesas/renesas_tsip_aes.c

```diff
@@ -0,0 +1,150 @@
+ * (at your option) any later version.
+ *
+ * wolfSSL is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1335, USA
+ */
+
+#include <string.h>
+#include <stdio.h>
+
+#ifdef HAVE_CONFIG_H
+    #include <config.h>
+#endif
+#include <wolfssl/wolfcrypt/settings.h>
+#include <stdio.h>
+
+#ifndef NO_AES
+
+#if defined(WOLFSSL_RENESAS_TSIP_CRYPT) && \
+    !defined(NO_WOLFSSL_RENESAS_TSIP_CRYPT_AES)
+
+#include <wolfssl/wolfcrypt/wc_port.h>
+#include <wolfssl/wolfcrypt/error-crypt.h>
+
+#include <wolfssl/wolfcrypt/aes.h>
+#include "wolfssl/wolfcrypt/port/Renesas/renesas-tsip-crypt.h"
+
+struct Aes;
+
+int wc_tsip_AesCbcEncrypt(struct Aes* aes, byte* out, const byte* in, word32 sz)
+{
+    tsip_aes_handle_t _handle;
+    word32 ret;
+    word32 blocks = (sz / AES_BLOCK_SIZE);
+    uint32_t dataLength;
+    byte *iv;
+    
+    /* while doing TLS handshake, TSIP driver keeps true-key and iv on the device */
+    /* iv is dummy */
+    iv = (uint8_t*)aes->reg;
+    
+    if ((in == NULL) || (out == NULL) || (aes == NULL))
+      return BAD_FUNC_ARG;
+    
+    if((ret = tsip_hw_lock()) != 0){
+        WOLFSSL_MSG("Failed to lock");
+        return ret;
+    }
+    
+    if (aes->ctx.keySize == 16) {
+        ret = R_TSIP_Aes128CbcEncryptInit(&_handle, &aes->ctx.tsip_keyIdx, iv);
+    } else if (aes->ctx.keySize == 32) {
+        ret = R_TSIP_Aes256CbcEncryptInit(&_handle, &aes->ctx.tsip_keyIdx, iv);
+    } else {
+        tsip_hw_unlock();
+        return -1;
+    }
+    
+    while (ret == TSIP_SUCCESS && blocks--) {
+        
+        if (aes->ctx.keySize == 16)
+            ret = R_TSIP_Aes128CbcEncryptUpdate(&_handle, (uint8_t*)in, 
+                                    (uint8_t*)out, (uint32_t)AES_BLOCK_SIZE);
+        else
+            ret = R_TSIP_Aes256CbcEncryptUpdate(&_handle, (uint8_t*)in, 
+                                    (uint8_t*)out, (uint32_t)AES_BLOCK_SIZE);
+        
+        in  += AES_BLOCK_SIZE;
+        out += AES_BLOCK_SIZE;
+    }
+    
+    if (ret == TSIP_SUCCESS) {
+        if (aes->ctx.keySize == 16) {
+            ret = R_TSIP_Aes128CbcEncryptFinal(&_handle, out, &dataLength);
+        } else {
+            ret = R_TSIP_Aes256CbcEncryptFinal(&_handle, out, &dataLength);
+        }
+    }
+    
+    tsip_hw_unlock();
+    return ret;
+}
+
+int wc_tsip_AesCbcDecrypt(struct Aes* aes, byte* out, const byte* in, word32 sz)
+{
+   tsip_aes_handle_t _handle;
+    word32 ret;
+    word32 blocks = (sz / AES_BLOCK_SIZE);
+    uint32_t dataLength;
+    byte *iv;
+    
+    iv = (uint8_t*)aes->reg;
```

Comment:
Like above, this iv assignment should move below NULL checks, in case "aes" is NULL.

Reply:
Done. Thanks!